specs/IFramePermissionRequested.md

```diff
@@ -0,0 +1,434 @@
+                        message += "Do you want to grant permission?\n";
+                        message +=
+                            (permissionArgs.IsUserInitiated
+                                ? "This request came from a user gesture."
+                                : "This request did not come from a user gesture.");
+
+                        var selection = MessageBox.Show(message, "iframe PermissionRequest",
+                            MessageBoxButton.YesNoCancel);
+
+                        permissionArgs.State = CoreWebView2PermissionState.Default;
+
+                        if (selection == MessageBoxResult.Yes)
+                        {
+                            permissionArgs.State = CoreWebView2PermissionState.Allow;
+                            m_cachedPermissions[cachedKey] = true;
+                        }
+                        else if (selection == MessageBoxResult.No)
+                        {
+                            permissionArgs.State = CoreWebView2PermissionState.Deny;
+                            m_cachedPermissions[cachedKey] = false;
+                        }
+
+                        PutHandled(permissionArgs);
+                    }
+                }, null);
+            };
+        }
+        catch (NotImplementedException exception)
+        {
+            MessageBox.Show(this, "Frame Permission Requested Failed: " + exception.Message,
+                            "Frame Permission Requested");
+        }
+    };
+}
+
+string NameOfPermissionKind(CoreWebView2PermissionKind kind)
+{
+    switch (kind)
+    {
+        case CoreWebView2PermissionKind.Microphone:
+            return "Microphone";
+        case CoreWebView2PermissionKind.Camera:
+            return "Camera";
+        case CoreWebView2PermissionKind.Geolocation:
+            return "Geolocation";
+        case CoreWebView2PermissionKind.Notifications:
+            return "Notifications";
+        case CoreWebView2PermissionKind.OtherSensors:
+            return "Generic Sensors";
+        case CoreWebView2PermissionKind.ClipboardRead:
+            return "Clipboard Read";
+        default:
+            return "Unknown resources";
+    }
+}
+
+void PutHandled(CoreWebView2PermissionEventArgs args)
+{
+    // In the case of an iframe requesting permission, the default behavior is
+    // to first raise the PermissionRequested event off of the CoreWebView2Frame
+    // and invoke it's handlers, and then raise the event off the CoreWebView2
+    // and invoke it's handlers. However, If we set Handled to true on the
+    // CoreWebView2Frame event handler, then we will not raise the
+    // PermissionRequested event off the CoreWebView2.
+    try
+    {
+        // NotImplementedException could be thrown if underlying runtime did not
+        // implement Handled. However, we only run this code after checking if
+        // CoreWebView2Frame.PermissionRequested exists, and both exist together,
+        // so it would not be a problem.
```

Comment:
The comment is explaining why the try/catch isn't necessary, so why do we do it?

Reply:
Remove try/catch, keep updated comment.